src/charm.py

```diff
@@ -198,6 +243,55 @@ def _generate_cu_config(self) -> str:
             tac=self._charm_config.tac,
         )
 
+    @staticmethod
+    def _generate_network_annotations() -> List[NetworkAnnotation]:
+        """Generate a list of NetworkAnnotations to be used by CU's StatefulSet.
+
+        Returns:
+            List[NetworkAnnotation]: List of NetworkAnnotations
+        """
+        return [
+            NetworkAnnotation(
+                name=N3_NETWORK_ATTACHMENT_DEFINITION_NAME,
+                interface=N3_INTERFACE_NAME,
+            )
+        ]
+
+    def _get_nad_base_config(self) -> dict:
+        return {
+            "cniVersion": "0.3.1",
+            "type": "macvlan",
+            "ipam": {
+                "type": "static",
+                "addresses": [
+                    {
+                        "address": self._get_n3_ip_address_from_config(),
+                    }
+                ],
```

Comment:
Looking at the upstream, we should also have a possibility of defining routes here.

Reply:
Added.